cms/static/js/views/move_xblock_list.js

```diff
@@ -0,0 +1,201 @@
+                    {
+                        xblocks: this.childrenInfo.children,
+                        noChildText: this.getNoChildText(),
+                        categoryText: this.getCategoryText(),
+                        parentDisplayname: this.parentInfo.parent.get('display_name'),
+                        XBlocksCategory: this.childrenInfo.category,
+                        currentLocationIndex: this.getCurrentLocationIndex()
+                    }
+                )
+            );
+            Backbone.trigger('move:childrenRendered', this.breadcrumbInfo());
+            return this;
+        },
+
+        /**
+         * Forward button press handler. This will render all the childs of an XBlock.
+         *
+         * @param {Object} event
+         */
+        renderChildren: function(event) {
+            this.renderXBlockInfo(
+                'forward',
+                $(event.target).closest('.xblock-item').data('itemIndex')
+            );
+        },
+
+        /**
+         * Breadcrumb button press event handler. Render all the childs of an XBlock.
+         *
+         * @param {any} newParentIndex  Index of a parent XBlock
+         */
+        handleBreadcrumbButtonPress: function(newParentIndex) {
+            this.renderXBlockInfo('backward', newParentIndex);
+        },
+
+        /**
+         * Render XBlocks based on `forward` or `backward` navigation.
+         *
+         * @param {any} direction           `forward` or `backward`
+         * @param {any} newParentIndex      Index of a parent XBlock
+         */
+        renderXBlockInfo: function(direction, newParentIndex) {
+            if (direction === undefined) {
+                this.parentInfo.parent = this.model;
+            } else if (direction === 'forward') {
+                // clicked child is the new parent
+                this.parentInfo.parent = this.childrenInfo.children[newParentIndex];
+            } else if (direction === 'backward') {
+                // new parent will be one of visitedAncestors
+                this.parentInfo.parent = this.visitedAncestors[newParentIndex];
+                // remove visited ancestors
+                this.visitedAncestors.splice(newParentIndex);
+            }
+
+            this.visitedAncestors.push(this.parentInfo.parent);
+
+            if (this.parentInfo.parent.get('child_info')) {
+                this.childrenInfo.children = this.parentInfo.parent.get('child_info').children;
+            } else {
+                this.childrenInfo.children = [];
+            }
+
+            this.setDisplayedXBlocksCategories();
+            this.render();
+        },
+
+        /**
+         * Set parent and child XBlock categories.
+         */
+        setDisplayedXBlocksCategories: function() {
+            this.parentInfo.category = XBlockUtils.getXBlockType(
+                this.parentInfo.parent.get('category'),
+                this.visitedAncestors[this.visitedAncestors.length - 2]
```

Comment:
There is no mystery here. I just wanted to improve this calculation but didn't found an alternative.